src/main.rs

```diff
@@ -92,29 +108,51 @@ impl HanoiError {
 ///    * `Action`: if the input was well formed
 ///    * `Hanoi::UnknownCommand`: otherwise
 fn parse_action(input: &str) -> Result<Action,HanoiError> {
-    unimplemented!()
+    if input == "q" {
+      return Ok(Action::Quit);
+    }
+    let mut chars = input.chars();
+    let from = chars.next().and_then(char_to_peg);
+    let to = chars.next().and_then(char_to_peg);
+    match (from, to, chars.next()) {
+        (Some(from_peg), Some(to_peg), None) => Ok(Action::Move(Move::new(from_peg, to_peg))),
```

Comment:
This is awesome. Jackson and I are looking at how people did parsing and I really like your `match` on a 3-tuple.